ramhorns/src/template/parse.rs

```diff
@@ -96,8 +100,23 @@ impl<'tpl> Template<'tpl> {
                                 }
 
                                 return Ok(self.blocks.len() - blocks_at_start);
-                            },
-                            _ => {},
+                            }
+                            Tag::Partial => {
+                                let path = dir.join(name);
+                                if !partials.contains_key(name) {
+                                    let template = Template::load(
```

Comment:
I'm not sure I'm okay with this. Loading resources from fs should not be done within the parser, this could be extra dangerous if templates are user provided as we don't sanitize names here, so if you ever were to expose this on a web server to users, it could leak sensitive data by the virtue of doing `{{>../../../somefile}}`.

Reply:
A solution that just occured to me would be to use a generic here. Create a trait called `Partials` with a single method on it:

```rust
fn get_partial_blocks(&mut self, name: &str) -> Result<&[Block], Error>;
```

You can then have 2 separate implementations for it:

1. Always return an `Error` for when you parse a template from a `String` or `&str`.
2. Existing behaviour: load partials from disc with a local cache when loading a template from file.

There should be some sanitization here so that you cannot go up in the folder structure.